### PR TITLE
Gh-356: Detail breaking changes

### LINKED ISSUE #356: Add details to breaking changes
In the [Gaffer 2 changelist page](https://github.com/gchq/gaffer-doc/blob/v2docs/docs/gaffer2.0/changelist.md) we should specifiy if the breaking changes were just to internal code (relevant to developers), will require changes to deployments (relevant to admins), or broke the outwards facing api (relevant to users). It would also be useful if we mentioned which alpha added the breaking change

### PR DESCRIPTION
**Pull request by t92549**
# Related Issue

- Resolve #356

### CODE REVIEW

File: docs/change-notes/changelist/v2-changes.md [hunk 1]
```diff
@@ -2,37 +2,43 @@
...
 ### Cache Improvements and fixes
 All "caches" within Gaffer received a lot of bug fixes which should make them significantly more stable and consistent over time. This should improve usability of FederatedStores, NamedOperations and NamedViews.
-!!! danger "Breaking change"
+!!! danger "Breaking change | admin | [alpha-0.6](https://github.com/gchq/Gaffer/releases/tag/gaffer2-2.0.0-alpha-0.6)"
     The cache will need to be reloaded, as the new internal cache interface has changed.  To do this, export all of the contents of your cache, upgrade, then re-add everything manually.
 
 ### Removal of Deprecated code
 All of Gaffer 1's deprecated code has been removed.
-!!! danger "Breaking change"
+!!! danger "Breaking change | user | [alpha-0.1](https://github.com/gchq/Gaffer/releases/tag/gaffer2-2.0.0-alpha-0.1)"
```
Comment (GCHQDeveloper314) [inline docs/change-notes/changelist/v2-changes.md:28]: Shouldn't this be "developer" or both of these?

Comment (t92549) [reply docs/change-notes/changelist/v2-changes.md:28]: I tried to make it like a hierarchy, where "user" actually means "user, admin and developer". I tried to describe it at the top but I don't think it's very clear.

Comment (GCHQDeveloper314) [reply docs/change-notes/changelist/v2-changes.md:28]: I see. In that section you could put "In addition to the above," at the start of the lines for admin and dev?

Comment (t92549) [reply docs/change-notes/changelist/v2-changes.md:28]: Have explained it clearly, what do you think @GCHQDeveloper314?

Comment (GCHQDeveloper314) [reply docs/change-notes/changelist/v2-changes.md:28]: It's clearer now.

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**